samples/inputs/dropdown/position/src/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import { IgrDropdown, IgrButton, IgrDropdownItem, IgrDropdownModule, IgrButtonModule, IgrDropdownItemModule, DropdownPlacement } from 'igniteui-react';
+import 'igniteui-webcomponents/themes/light/bootstrap.css';
+
+IgrDropdownModule.register();
+IgrDropdownItemModule.register();
+IgrButtonModule.register();
+
+export default class DropDownPosition extends React.Component<any, any> {
+
+    public dropdownRef: IgrDropdown;
+
+    constructor(props: any) {
+        super(props);
+        this.onDropDownRef = this.onDropDownRef.bind(this);     
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="container sample center">
+                <IgrDropdown ref={this.onDropDownRef} distance="5" change={(e)=>this.onChange(e)} placement="bottom">
+                    <div slot="target">
+                        <IgrButton><span>Options</span></IgrButton>
+                    </div>
+                    <IgrDropdownItem><span>top</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>top-start</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>top-end</span></IgrDropdownItem>
+                    <IgrDropdownItem selected><span>bottom</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>bottom-start</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>bottom-end</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>right</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>right-start</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>right-end</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>left</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>left-start</span></IgrDropdownItem>
+                    <IgrDropdownItem><span>left-end</span></IgrDropdownItem>
+                </IgrDropdown>
+            </div>
+        );
+    }
+
+    public onDropDownRef(dropdown: IgrDropdown){
+        if (!dropdown) { return; }
+        this.dropdownRef = dropdown;
+        this.setState({});
+    }
+
+    public onChange(event: any): void {
+        if(this.dropdownRef){
+            for (let i = 1; i < event.i.nativeElement.children.length; i++) {
+                let item = event.i.nativeElement.children[i];
+                if (item.selected){
+                    this.dropdownRef.placement = item.value;
```

Comment:
This results in a crash for hyphenated options. The members of DropdownPlacement enum are all concatenated, meaning we should either set this.dropdownRef.placement to a concatenated value (e.g. 'topstart' instead of 'top-start') or to the numeric counterpart of the enum member (in the current implementation of onChange() this could be just 'i - 1')

A whole different issue, however, is the actual positioning of the dropdown for some of the options. It seems weird, especially compared to the same behavior in Web Components. This might need to be logged as a work item and investigated.

Reply:
I changed the values of the items to reflect the values that should be set to the placement property of the dropdown component that is, the concatenated values ('topstart') instead of the hyphenated ones ('top-start').

Reply:
I also logged the described behavior with concatenated DropdownPlacement enum options not working as expected and not setting the dropdown placement as a work item [28903](https://infragistics.visualstudio.com/NetAdvantage/_workitems/edit/28903).